aptos-move/framework/aptos-framework/sources/object.move

```diff
@@ -420,6 +425,35 @@ module aptos_framework::object {
         owner(object) == owner
     }
 
+    /// Return true if the provided address has indirect or direct ownership of the provided object.
+    public fun owns<T: key>(object: Object<T>, owner: address): bool acquires ObjectCore {
```

Comment:
If there is a cycle and `owner` is any node in that cycle, it is will always return `true`?

Reply:
If it is less than 8